src/modules/new-request-form/fields/LookupField.tsx

```diff
@@ -0,0 +1,216 @@
+  const [isFirstLoad, setIsFirstLoad] = useState<boolean>(true);
+  const { t } = useTranslation();
+
+  const customObjectKey = getCustomObjectKey(
+    relationship_target_type as string
+  );
+
+  const loadingOption = {
+    name: t("new-request-form.lookupfield.loading-options", "Loading items..."),
+    id: "loading",
+  };
+
+  const noResultsOption = {
+    name: t(
+      "new-request-form.lookupfield.no-matches.found",
+      "No matches found"
+    ),
+    id: "no-results",
+  };
+
+  const fetchSelectedOption = useCallback(
+    async (selectionValue: string) => {
+      const res = await fetch(
+        `/api/v2/custom_objects/${customObjectKey}/records/${selectionValue}`
+      );
+      const { custom_object_record } = await res.json();
+      const newSelectedOption = {
+        name: custom_object_record.name,
+        value: custom_object_record.id,
+      };
+      setSelectedOption(newSelectedOption);
+      setInputValue(custom_object_record.name);
+      onChange(custom_object_record.id);
+    },
+    [customObjectKey, onChange]
+  );
+
+  const handleChange = useCallback<NonNullable<IComboboxProps["onChange"]>>(
+    async ({ inputValue, selectionValue }) => {
+      setIsFirstLoad(false);
+      if (selectionValue !== undefined) {
+        if (selectionValue == "") {
+          setSelectedOption(EMPTY_OPTION);
+          setInputValue(EMPTY_OPTION.name);
+          onChange(EMPTY_OPTION.value);
+        } else {
+          await fetchSelectedOption(selectionValue as string);
+        }
+        return;
+      }
+
+      if (inputValue !== undefined) {
+        setInputValue(inputValue);
+        if (inputValue === "") {
+          setOptions([]);
+          setSelectedOption(null);
+          onChange("");
+        } else {
+          const searchParams = new URLSearchParams();
+          searchParams.set("name", inputValue.toLocaleLowerCase());
+          searchParams.set("source", "zen:ticket");
+          searchParams.set("field_id", fieldId.toString());
+          searchParams.set("user_id", userId.toString());
+
+          setIsLoadingOptions(true);
+          try {
+            const response = await fetch(
+              `/api/v2/custom_objects/${customObjectKey}/records/autocomplete?${searchParams.toString()}`
+            );
+
+            const data = await response.json();
+            if (data !== undefined) {
+              setOptions(
+                data.custom_object_records.map(
+                  ({ name, id }: { name: string; id: string }) => ({
+                    name,
+                    value: id,
+                  })
+                )
+              );
+            } else {
+              setOptions([]);
+            }
+          } catch (error) {
+            return error;
+          } finally {
+            setIsLoadingOptions(false);
+          }
+        }
+      }
+      return;
+    },
+    [customObjectKey, fetchSelectedOption]
```

Comment:
According to ESLint, we should add more things to the dependencies array:
> React Hook useCallback has missing dependencies: 'fieldId', 'onChange', and 'userId'. Either include them or remove the dependency array. If 'onChange' changes too often, find the parent component that defines it and wrap that definition in useCallback.

And I guess at this point we should wrap the `handleChange` in the `NewRequestForm` component with a `useCallback` as well

Reply:
Ok I will try this